### PR TITLE
Adding fugue integration and tests

### PR DESCRIPTION
**Pull request by nils-braun**
After the great conversation with @goodwanghan and @kvnkho, I am implementing an integration of dask-sql into fugue using basically the code written by the two fugue owners. It allows to replace the logic of SQL handling in fugue by dask-sql. 

So far, dask-sql seems to be faster (with especially larger data I guess), but fugue has a larger set of understood SQL. However I hope, that dask-sql will be en par quickly.

Still to do for this PR: documentation

### CODE REVIEW

**Comment by kvnkho** [comment]
Hey, I can help you get started with the documentation. Maybe I can PR into that `docs` folder without generating the docs and then you can edit and place it where you see fit?

File: tests/integration/test_fugue.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import pytest
+
+import pandas as pd
+from pandas.testing import assert_frame_equal
+
+fugue_sql = pytest.importorskip("fugue_sql")
+# needs to be imported after the check for fugue
+from dask_sql.integrations.fugue import DaskSQLExecutionEngine
+
+
```
Comment (kvnkho) [inline tests/integration/test_fugue.py:10]: Awesome!

**Comment by nils-braun** [comment]
That would be awesome!

**Comment by nils-braun** [comment]
Note: The test error will hopefully be fixed in #117

**Comment by goodwanghan** [comment]
This is amazing, thank you @nils-braun

**Comment by nils-braun** [comment]
Thanks to you for your hard work and nice preparation! If I can help you with the docs, feel free to contact me anytime

**Comment by goodwanghan** [comment]
Hey @nils-braun can you give the access to submit my pr?

I made some changes to this pr to add a simple function `fsql` to utilize your Context object directly

```python
def test_fsql():
    def assert_eq(df:pd.DataFrame) -> None:
        assert_frame_equal(df, pd.DataFrame({"a": [1]}))

    # the simplest case: the SQL does not use any input and does not generate output
    fsql("""
    CREATE [[0],[1]] SCHEMA a:long
    SELECT * WHERE a>0
    OUTPUT USING assert_eq
    """)

    # it can directly use the dataframes inside dask-sql Context
    c = Context()
    c.create_table(
        "df", dd.from_pandas(pd.DataFrame([[0],[1]],
        columns=["a"]), npartitions=2))

    fsql("""
    SELECT * FROM df WHERE a>0
    OUTPUT USING assert_eq
    """, c)

    # for dataframes with name, they can register back to the Context (register=True)
    # the return of fsql is the dict of all dask dataframes with explicit names
    result = fsql("""
    x=SELECT * FROM df WHERE a>0
    OUTPUT USING assert_eq
    """, c, register=True)
    assert isinstance(result["x"], dd.DataFrame)
    assert "x" in c.tables

    # integration test with fugue transformer extension
    c = Context()
    c.create_table(
        "df1", dd.from_pandas(pd.DataFrame([[0,1],[1,2]],
        columns=["a", "b"]), npartitions=2))
    c.create_table(
        "df2", dd.from_pandas(pd.DataFrame([[1,2], [3,4],[-4,5]],
        columns=["a", "b"]), npartitions=2))

    # schema: *
    def cumsum(df:pd.DataFrame) -> pd.DataFrame:
        return df.cumsum()

    fsql("""
    SELECT * FROM df1 WHERE a>0 UNION ALL SELECT * FROM df2 WHERE a>0 PERSIST
    result1 = TRANSFORM PREPARTITION BY a PRESORT b USING cumsum
    result2 = TRANSFORM PREPARTITION BY b PRESORT a USING cumsum
    PRINT result1, result2
    """, c, register=True)
    assert "result1" in c.tables
    assert "result2" in c.tables

```

**Comment by nils-braun** [comment]
As discussed with @goodwanghan on slack, I will merge now and the additions by him will come afterwards

**Pull request merged**